design/welcome-email.feature

```diff
@@ -0,0 +1,93 @@
+Feature: Welcome new subscribers
+  - As a store owner
+  - I want to send welcome emails
+  - So that I will connect with my new subscribers immediately
+
+  Entities:
+    Guest - A person who has created an order or signed up for a newsletter. Guest has not created
+      unique customer account, but has a distinct email address in Magento store.
+    Customer - A person who has created an account in store.
+    Newsletter Subscriber - Person who exists in newsletter subscribers list in Magento. Newsletter
+      subscriber can have subscribed and unsubscribed status.
+    Need to Confirm - A setting in Magneto store that requires subscribers to confirm their
+      subscription before they can receive newsletters.
+
+  Background:
+    Given store owner has created opt-in automation in Smaily
+
+  Rule: Welcome email is sent to new subscribers
+    Scenario: Guest subscribes
+      Given Guest fills newsletter sign-up form
+      When guest submits the form
+      Then welcome email is sent
+
+    Scenario: Guest subscribes with an email address belonging to Newsletter Subscriber
+      Given Guest fills newsletter sign-up form
+      And the same email address belongs to Newsletter Subscriber
+      When Guest submits the form
+      Then no welcome email is sent
+
+    Scenario: Guest subscribes with an email address belonging to Customer
+      Given Guest fills newsletter sign-up form
+      And the same email address already belongs to a Customer
+      When Guest submits the form
+      Then an error message is shown
+      And no welcome email is sent
+
+    Scenario: Customer subscribes
+      Given Customer fills newsletter sign-up form
+      When Customer submits the form
+      Then welcome email is sent
```

Comment:
When reading this scenario I am not sure what Customers are affected by it. Does any Customer who submits the form get a welcome email?